### PR TITLE
fix: sidecars download permissions TDE-1007

### PR DESCRIPTION
**Pull request by amfage**
#### Motivation

Simplify downloading of sidecar files and overcoming the head_object permission loading issue.

#### Modification

The previous code checked if a file existed and if so, read it. It would not load credentials if access was denied. This added an unneccessary step as well as failing to load sidecar files due to permissions issues. This modification checks for sidecar files by reading them. If they do not exist, that is logged but does not cause an error.

#### Checklist

_If not applicable, provide explanation of why._

- [x] Tests updated
- [x] Docs updated
- [x] Issue linked in Title

### CODE REVIEW

File: scripts/translate_ascii.py [hunk 1]
```diff
@@ -55,12 +55,13 @@ def main() -> None:
         get_log().info("ascii file translation complete", duration=time_in_ms() - start_time, count=len(tiffs))
 
         # copy any sidecar files to target
-        sidecars = []
-        start_time = time_in_ms()
-        for ls in asc_files:
-            sidecars += find_sidecars(ls, [".prj", ".tfw"])
-        write_all(inputs=sidecars, target=arguments.target)
-        get_log().info("sidecar files copied", duration=time_in_ms() - start_time, count=len(sidecars))
+        # TODO: TDE-1007 update translate_ascii for new sidecar download method
```
Comment (l0b0) [inline scripts/translate_ascii.py:58]: This presumably still needs doing?

Comment (amfage) [reply scripts/translate_ascii.py]: Yes, I put it back in draft as I realised that but you were v. quick :)

**Comment by paulfouquet** [comment]
Is this still work in progress?

File: scripts/files/fs.py [hunk 2]
```diff
@@ -77,40 +81,34 @@ def write_all(inputs: List[str], target: str, concurrency: Optional[int] = 4) ->
...
-        raise Exception("Not all source files were written")
+        raise Exception("Not all mandatory source files were written")
+
     return written_tiffs
 
 
-def find_sidecars(inputs: List[str], extensions: List[str], concurrency: Optional[int] = 4) -> List[str]:
-    """Searches for sidecar files.
-     A sidecar files is a file with the same name as the input file but with a different extension.
+def write_sidecars(inputs: List[str], target: str, concurrency: Optional[int] = 4) -> None:
```
Comment (paulfouquet) [inline scripts/files/fs.py:89]: Might be a good idea to add some tests

File: .pylintrc [hunk 3]
```diff
@@ -1,5 +1,6 @@
 [MASTER]
 disable =
+        duplicate-code,
```
Comment (paulfouquet) [inline .pylintrc:3]: As discussed, it would be better to ignore this warning in specific places

File: scripts/standardising.py [hunk 4]
```diff
@@ -135,8 +135,12 @@ def standardising(
     # Download any needed file from S3 ["/foo/bar.tiff", "s3://foo"] => "/tmp/bar.tiff", "/tmp/foo.tiff"
     with tempfile.TemporaryDirectory() as tmp_path:
         standardized_working_path = os.path.join(tmp_path, standardized_file_name)
-        sidecars = find_sidecars(files.inputs, [".prj", ".tfw"])
-        source_files = write_all(files.inputs + sidecars, f"{tmp_path}/source/")
+        sidecars: List[str] = []
+        for extension in [".prj", ".tfw"]:
```
Comment (paulfouquet) [inline scripts/standardising.py:139]: NIT: could store this list of sidecar files extensions in a constant

**Review by paulfouquet — APPROVED** [review]
Just a NIT comment apart from that looks good to me

**Review by MDavidson17 — APPROVED** [review]

**Pull request merged**